### PR TITLE
feat: basic annotation syntax support for lang

### PR DESCRIPTION
**Pull request by sid-bruno**
### Description

[JIRA](https://usebruno.atlassian.net/browse/BRU-2991)

Adds `@annotation` / `@annotation('value')` decorator syntax to `.bru` dictionary blocks. The Ohm grammar is extended with annotation rules (`annotation`, `annotationentry`, `pairannotations`) across `bruToJson.js`, `collectionBruToJson.js`, and `envToJson.js`. Annotations are attached via a hidden `Symbol` key and exposed as an `annotations` array on parsed pairs — only when present, preserving backward compatibility. Serializers (`jsonToBru.js`, `jsonToCollectionBru.js`, `jsonToEnv.js`) gain a `serializeAnnotations()` helper that handles no-arg, single-line (auto-quoting), and multiline (`'''`) cases, applied across all dictionary block paths. A comprehensive test suite (`annotations.spec.js`) covers parsing, serialization, and roundtrip correctness.

```bru
headers {
  @description('Auth token') authorization: Bearer {{token}}
  @required
  @string content-type: application/json
  @deprecated ~x-old-header: legacy
}

params:query {
  @description('Search term') q: hello
  @string ~limit: 10
}

assert {
  @description('Status check') res.status: eq 200
}

vars:pre-request {
  @description('Base API URL') baseUrl: https://api.example.com
  @deprecated ~oldVar: unused
}
```

#### Contribution Checklist:

- [ ] **I've used AI significantly to create this pull request**
- [ ] **The pull request only addresses one issue or adds one feature.**
- [ ] **The pull request does not introduce any breaking changes**
- [ ] **I have added screenshots or gifs to help explain the change if applicable.**
- [ ] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [ ] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

#### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

## Summary by CodeRabbit

* **New Features**
  * Added decorator-style annotations (e.g., @name, @name(value), @name('''...''')) that can be placed before pair keys across params, headers, form/multipart fields, vars, and asserts; parsed outputs include an annotations field when present, and serialization emits annotations (preserving formatting for disabled items and selecting quotes to avoid escapes).

* **Tests**
  * Added comprehensive tests and fixture covering parsing, serialization, quoting, multiline/special-character edge cases, placement variants, disabled pairs, and roundtrip equality.

### CODE REVIEW

**Comment by sid-bruno** [comment]
@coderabbitai review

**Comment by sid-bruno** [comment]
### Code review

Found 2 issues:

1. `assertpair` grammar rule does not include `pairannotations`, but `jsonToBru` calls `serializeAnnotations` for assertion items — annotations on assert pairs are written to `.bru` but silently dropped on re-parse (round-trip data loss)

https://github.com/usebruno/bruno/blob/22569e229184d65414ef786d4ae8afb7a5bf61fc/packages/bruno-lang/v2/src/bruToJson.js#L89-L92

https://github.com/usebruno/bruno/blob/22569e229184d65414ef786d4ae8afb7a5bf61fc/packages/bruno-lang/v2/src/jsonToBru.js#L730-L741

2. `.filter(Boolean)` in `annotationmultilinetextblock` removes empty strings from the mapped lines array, which silently drops internal blank lines in multiline annotation values — a value like `"first\n\nsecond"` round-trips back as `"first\nsecond"`. Removing `.filter(Boolean)` (while keeping the empty-string mapping for blank lines) would fix this, matching the approach used in the `multilinetextblock` handler.

https://github.com/usebruno/bruno/blob/22569e229184d65414ef786d4ae8afb7a5bf61fc/packages/bruno-lang/v2/src/bruToJson.js#L394-L400

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Comment by sid-bruno** [comment]
Feedback

- Remove inline annotation decorators

**Comment by sid-bruno** [comment]
> Feedback
> 
>     * Remove inline annotation decorators

Updated, added limited tests to make sure that it throws if the decorators/annotations are inline

File: packages/bruno-lang/v2/src/jsonToBru.js [hunk 1]
```diff
@@ -6,6 +6,22 @@ const jsonToExampleBru = require('./example/jsonToBru');
 const enabled = (items = [], key = 'enabled') => items.filter((item) => item[key]);
 const disabled = (items = [], key = 'enabled') => items.filter((item) => !item[key]);
 
+const serializeAnnotations = (annotations) => {
```
Comment (bijin-bruno) [inline packages/bruno-lang/v2/src/jsonToBru.js]: I think this should be moved to utils
